### ISSUE TITLE
Rethrowing an exception within the catch bloc is not detected when it's done by a method

### ISSUE THREAD

**Issue by bilougit** [thecodingmachine/phpstan-strict-rules#54]
```php
try {
// ....
} catch (\Throwable $e) {
     // ....
     \MyProject\Namespace\MyCustomException::raise($id, $e /*will be injected as previous*/)
}

// src/Exception/MyCustomException.php
namespace MyProject\Namespace;

class MyCustomException extends \Exception
{
    public static function raise(string $id, \Throwable $previous)
    {
        throw new self("something went wrong with #{$id}", 0, $previous);
    }
}
```

This will not be detected and I get this message:

`caught "Throwable" must be rethrown. Either catch amore specific exception or add a "throw" clause in the "catch" block to propagate the exception. More info:`

Does this rule require the rethrow be explicit or is it just a small bug?

**Comment by homersimpsons** [comment]
> Does this rule require the rethrow be explicit or is it just a small bug?

I consider that the rethrow should be explicit. Hence we will not weaken the rule with this use case and I instead recommend (https://github.com/thecodingmachine/phpstan-strict-rules/pull/55#issuecomment-4822591113):
```php
try {
// ....
} catch (\Throwable $e) {
     // ....
     throw \MyProject\Namespace\MyCustomException::create($id, $e)
}
```

Thanks for your time creating this issue and a pull request implementing it.
